### ISSUE TITLE
Numpy operator "/" not supported for types "floating[Any]" and "floating[Any]"

### ISSUE THREAD

**Issue by Antyos** [microsoft/pylance-release#5393]
## Environment data

-   Language Server version: 2023.12.1
-   OS and version: Windows 10 22H2
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.12.1
-   Numpy version 1.26.3

## Code Snippet

Snippet:
```python
import numpy as np

a = np.arange(4)

x = a - np.mean(a)  # OK
y = a - np.std(a)   # OK
z = np.std(a) / np.mean(a)  # BAD
#   ^^^^^^^^^^^^^^^^^^^^^^  Operator "/" not supported for types "floating[Any]" and "floating[Any]"
#                           Pylance (reportGeneralTypeIssues)
```

## Expected behavior

The standard deviation and mean of a numerical array will are both numerical, so dividing them shouldn't be an issue.

## Actual behavior

There is no error when operating on the original array with its mean or standard deviation. However, for some reason combining both results in an error. Both functions seem to return some type of `Any`. At a basic level, I understand why `Any / Any` can result in an error, but it seems silly here, especially because both values are recognized to be `floating`.

Please let me know if this issue is better off filed in Numpy's repo.

## Logs

See here: [logs.txt](https://github.com/microsoft/pylance-release/files/14042012/logs.txt)

**Comment by debonte** [comment]
This was fixed in [Pyright 1.1.337](https://github.com/microsoft/pyright/releases/tag/1.1.337).  I didn't track down the specific change/issue on the Pyright side. 

Pylance 2023.12.1 is based on Pyright 1.1.336 and therefore doesn't inlude the fix. But if you [switch to Pylance prerelease](https://github.com/microsoft/pylance-release/wiki/FAQ#how-to-switch-pylance-builds) you'll get a build with a more recent Pyright build where this is fixed. Any build 2023.12.100 or later will be fine.

**Comment by Antyos** [comment]
Oh, well that's an easy fix. Thanks!
